paimon-core/src/main/java/org/apache/paimon/iceberg/metadata/IcebergDataField.java

```diff
@@ -190,6 +199,70 @@ private static Object toTypeObject(DataType dataType, int fieldId, int depth) {
         }
     }
 
+    public DataType getDataType() {
```

Comment:
This class already has a `dataType` member. Check if `dataType` is null, if not just return that object, otherwise calculate data type from `type` string.